src/utils.jl

```diff
@@ -118,4 +118,22 @@ function householderbasis(n)
       u, v = v, u
   end
   u, v
-end
\ No newline at end of file
+end
+
+"""
+    mayberound(λ, x, tol)
+
+Round `λ` to `x` if it is within the tolerance `tol`.
+"""
+function mayberound(λ, x, tol) 
+  L = isapprox.(λ, x, atol = tol)
+  if any(L)
+    if sum(L) > 1
+      error("Tolerances too large for ")
+    else
+      return x[findfirst(L)]
+    end
+  else
+    return λ
+  end
+end
```

Comment:
This logic here is overly complicated and slow. We just need to treat the scalar case and use broadcast at the call site inside the intersection algorithms if necessary.

Reply:
I wanted to have the option to use
`mayberound(λ, [0,1], atol(T))`
instead of 
`mayberound(mayberound(λ, 0, atol(T)), 1, atol(T))`

but I guess a vectorized function is too complicated (and slow)

I now implemented the simple version: 
`mayberound(λ, x, tol) = isapprox(λ, x, atol = tol) ? x : λ`

Broadcasting does not work, but one can call the function iteratively.

Reply:
Yes, in terms of design it is better to keep the signature simple and call the function twice. One thing that you can consider if this is often used with 0 and 1 is to define a local function `maybe0or1` that calls `mayberound` twice. But I am still not sure how often we are planning to use `mayberound` to justify adding it here as a utility function.

Are you expecting to call this function more than 2 times in our intersection algorithms?

Reply:
in `segments.jl` `maybe0or1` is needed 4 times, but i can define `maybe0or1` locally there.

in `raysegment.jl` `mayberound` is needed 5 times or 3 times and 1 `maybe0or1`, but here it can be called recursivly.